source/ImageSet.cpp

```diff
@@ -210,18 +223,30 @@ void ImageSet::Add(string path)
 void ImageSet::ValidateFrames() noexcept(false)
 {
 	string prefix = "Sprite \"" + name + "\": ";
-	AddValid(framePaths[0], paths[0], prefix, false);
-	AddValid(framePaths[1], paths[1], prefix, true);
+	AddValid(framePaths[0], paths[0], prefix, false, false);
+	AddValid(framePaths[1], paths[1], prefix, true, false);
+	AddValid(framePaths[2], paths[2], prefix, false, true);
+	AddValid(framePaths[3], paths[3], prefix, true, true);
 	framePaths[0].clear();
 	framePaths[1].clear();
+	framePaths[2].clear();
+	framePaths[3].clear();
+
+	static auto dropPaths = [&](vector<string> &toResize, const string &specifier) {
+		if(toResize.size() > paths[0].size())
+		{
+			Logger::LogError(prefix + to_string(toResize.size() - paths[0].size())
+				+ " extra frames for the " + specifier + " sprite will be ignored.");
+			toResize.resize(paths[0].size());
+		}
+	};
 
 	// Drop any @2x paths that will not be used.
-	if(paths[1].size() > paths[0].size())
-	{
-		Logger::LogError(prefix + to_string(paths[1].size() - paths[0].size())
-				+ " extra frames for the @2x sprite will be ignored.");
-		paths[1].resize(paths[0].size());
-	}
+	dropPaths(paths[1], "@2x");
+	// Drop any mask paths that will not be used.
+	dropPaths(paths[2], "mask");
+	// Drop any @2x mask paths that will not be used.
+	dropPaths(paths[3], "@2x mask");
```

Comment:
And adjust the comment above, no need to repeat the same comment thrice.
```suggestion
	DropPaths(paths[1], "@2x");
	DropPaths(paths[2], "mask");
	DropPaths(paths[3], "@2x mask");
```

Reply:
done